### PR TITLE
Checkout module git dev

### PR DESCRIPTION
**Pull request by GDYendell**

### CODE REVIEW

File: dls_ade/dls_checkout_module.py [hunk 1]
```diff
@@ -1,51 +1,82 @@
...
 
+    Returns:
+        An ArgumentParser instance with relevant arguments
+    """
+    parser = ArgParser(usage)
     parser.add_argument("module_name", type=str, default="",
                         help="name of module to checkout")
     parser.add_argument("-b", "--branch", action="store", type=str, dest="branch",
-                        help="Checkout from a branch rather than from the trunk")
+                        help="Checkout a branch rather than master")
```
Comment (ulrikpedersen) [inline dls_ade/dls_checkout_module.py]: "Checkout a specific named branch rather than the default (master)"

File: dls_ade/dls_checkout_module.py [hunk 2]
```diff
@@ -56,33 +87,48 @@ def main():
...
+            if args.branch in branches:
+                vcs_git.checkout_remote_branch(args.branch, repo)
+            else:
+                # Invalid branch name, print branches and ask user to
+                # re-enter the branch they want
+                print("Branch '" + args.branch + "' does not exist in " + source +
+                      "\nBranch List:\n")
+                for entry in branches:
+                    print(entry)
+                print("\nWhich branch would you like to checkout?")
```
Comment (ulrikpedersen) [inline dls_ade/dls_checkout_module.py]: Just bail out here if the requested branch does not exist. It's a good idea to print existing branches - but don't give the user a second attempt. They can run the program again with a modified branch argument...

If this application is ever used in an automated script it is practical to have an interactive state like that.

File: dls_ade/dls_list_modules.py [hunk 3]
```diff
@@ -3,33 +3,57 @@
...
+    """
     split_list = vcs_git.get_repository_list()
-    for path in split_list:
-        if source in path:
-            print(path)
+    prefix = len("controls/" + area) + 1
+    print("Modules in " + area + ":\n")
+    for module in split_list:
+        if source in module:
+            print(module[prefix:])
```
Comment (ulrikpedersen) [inline dls_ade/dls_list_modules.py]: I think this extraction of the module name from the items in the split_list could probably be done a bit more elegant? The whole prefix = len(...) + 1 is a bit worrying... How about splitting by "/", using os.path or maybe a reg-exp?

File: dls_ade/dls_release.py [hunk 4]
```diff
@@ -4,33 +4,42 @@
...
 
-# set default variables
 log_mess = "%s: Released version %s. %s"
 
 
 def make_parser():
-    ''' helper method containing arguments and help text '''
-
+    """
+    Takes default parser and adds module_name, release, --branch, --force, --no-test-build,
```
Comment (ulrikpedersen) [inline dls_ade/dls_release.py]: I think these can be formatted as a bullet list which makes it more readable in the generated docs...

File: dls_ade/vcs_git.py [hunk 5]
```diff
@@ -21,28 +29,30 @@ def is_repo_path(server_repo_path):
...
+    """
+    Returns formatted list of entries from 'ssh dascgitolite@dasc-git.diamond.ac.uk expand controls' command
 
+    Returns:
+        list: Reduced 'expand controls' output
+    """
     list_cmd = "ssh " + GIT_ROOT + " expand controls"
     list_cmd_output = subprocess.check_output(list_cmd.split())
-    split_list = list_cmd_output.split()
+    split_list = []
```
Comment (ulrikpedersen) [inline dls_ade/vcs_git.py]: Not really a requirement, but a nice to have for the next developer who comes along: A comment, describing what the output of the list_cmd looks like (or just a few lines of example output in the comment) and why it is parsed in the way you do...

File: dls_ade/vcs_git.py [hunk 6]
```diff
@@ -21,28 +29,30 @@ def is_repo_path(server_repo_path):
...
 def temp(area, module):
+    # >>> This has been properly implemented on start-new-module branch
 
     # >>> adjust for technical area
     if area == "ioc":
         pass
 
     path = "./"
-    # target = "git@github.com:GDYendell/git_test.git"
     target = "ssh://dascgitolite@dasc-git.diamond.ac.uk/controls/" + area + module
```
Comment (ulrikpedersen) [inline dls_ade/vcs_git.py]: Isn't there a constant defined somewhere else with "dascgitolite@dasc-git.diamond.ac.uk"?

**Comment by ulrikpedersen** [comment]
I've had a first got at reviewing this - see above comments. We can merge this when the few above-mentioned issues has been addressed. :+1:

**Comment by GDYendell** [comment]
According to this[http://matplotlib.org/sampledoc/cheatsheet.html], "* " will format to a bullet point in the html, but it doesn't seem to work for me.

**Pull request merged**